### PR TITLE
Change Vec to HashMap and allow the user to remove variables

### PR DESCRIPTION
**Pull request by felipeagc**
I was looking for some issues to tackle for hacktoberfest, hope you don't mind :)

Solves #2

### CODE REVIEW

**Comment by Freyskeyd** [comment]
hello @felipeagc,

Thank's for this! The PR is nice you just need to fix some style issue.

cc @epage @killercup for informations

**Review by killercup — APPROVED** [review]
LGTM!

File: src/lib.rs [hunk 1]
```diff
@@ -61,17 +65,43 @@ impl Environment {
...
+    /// use std::collections::HashMap;
+    ///
+    /// let e = environment::Environment::empty().insert("foo", "bar").remove("foo").compile();
+    ///
+    /// let mut hash_map: HashMap<OsString, OsString> = HashMap::new();
+    ///
+    /// assert_eq!(e, hash_map);
+    /// ```
+    pub fn remove<S: Into<OsString>>(mut self, key: S) -> Self {
+        self.vars.remove(&key.into());
```
Comment (epage) [inline src/lib.rs:99]: This won't let you remove inherited variables.

Whether in this PR or another, we should probably move `::std::env::vars_os().chain(self.vars)` into `fn inherit`

Comment (Freyskeyd) [reply src/lib.rs:99]: I was thinking about moving the var_os call at the instantiation. It avoid to have a `compile` method too.

**Comment by Freyskeyd** [comment]
bors try

**Comment by Freyskeyd** [comment]
Hello,

I need this modification to make the API evolve. I will close this PR but feel free to contribute again :)

thank's!

**Pull request closed without merging**